File: QuanLyPhongKham.Server/src/QuanLyPhongKham.Models/Entities/PHIEUKHAM.cs
using MayNghien.Infrastructures.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyPhongKham.Models.Entities
{
    public class PHIEUKHAM : BaseEntity
    {
        [Key]
        public Guid MaPK { get; set; }

        public DateTime NgayKham { get; set; }

        [MaxLength(500)]
        public string? TrieuChung { get; set; }

        [MaxLength(500)]
        public string? ChuanDoan { get; set; }

        [MaxLength(500)]
        public string? HuongDieuTri { get; set; }

        [MaxLength(30)]
        public string? TrangThaiTiepNhan { get; set; }

        [ForeignKey("LichHen")]
        public Guid? MaLH { get; set; }
        public LICHHEN? LichHen { get; set; }

        [ForeignKey("BacSi")]
        public Guid? MaBS { get; set; }
        public BACSI? BacSi { get; set; }

        public HOADON? HoaDon { get; set; }
        public ICollection<DONTHUOC>? DonThuocs { get; set; }
    }
}

File: QuanLyPhongKham.Client/src/models/PhieuKham.js
/**
 * @typedef {Object} PhieuKham
 * @property {string} maPK
 * @property {string|Date} ngayKham
 * @property {string} trieuChung
 * @property {string} chuanDoan
 * @property {string} huongDieuTri
 * @property {string} trangThaiTiepNhan
 * @property {string} maLH
 * @property {string} maBS
 * @property {string} tenBacSi
 * @property {string} tenBenhNhan
 */

export const createPhieuKham = (
  maPK = "",
  ngayKham = null,
  trieuChung = "",
  chuanDoan = "",
  huongDieuTri = "",
  trangThaiTiepNhan = "",
  maLH = null,
  maBS = null,
  tenBacSi = "",
  tenBenhNhan = "",
) => ({
  maPK,
  ngayKham,
  trieuChung,
  chuanDoan,
  huongDieuTri,
  trangThaiTiepNhan,
  maLH,
  maBS,
  tenBacSi,
  tenBenhNhan,
});

export const normalizePhieuKham = (item) => ({
  maPK: item?.maPK ?? item?.MaPK ?? "",
  ngayKham: item?.ngayKham ?? item?.NgayKham ?? null,
  trieuChung: item?.trieuChung ?? item?.TrieuChung ?? "",
  chuanDoan: item?.chuanDoan ?? item?.ChuanDoan ?? "",
  huongDieuTri: item?.huongDieuTri ?? item?.HuongDieuTri ?? "",
  trangThaiTiepNhan:
    item?.trangThaiTiepNhan ?? item?.TrangThaiTiepNhan ?? "",
  maLH: item?.maLH ?? item?.MaLH ?? null,
  maBS: item?.maBS ?? item?.MaBS ?? null,
  tenBacSi: item?.tenBacSi ?? item?.TenBacSi ?? "",
  tenBenhNhan: item?.tenBenhNhan ?? item?.TenBenhNhan ?? "",
});
